### DISCUSSION TITLE
Adding Resnames to the Universe

### DISCUSSION THREAD

**Discussion by rszukalo** [MDAnalysis/mdanalysis#4493] [MDAKit Q&A]
Hello,

I am trying to adapt MDAnalysis to my workflow and have run into an issue I was hoping for some assistance with.  I run simulations with potentials that don't explicitly define any topology in LAMMPS, thus they have no molecule information.  Thus far, I have been able to add a data file to my universe and manipulate it:

>> data_file = "Ih_60K.data"
>> u = mda.Universe(data_file, atom_style='id type x y z')  

When I get information about this universe it reads as expected:

>> print('Atoms: ', u.atoms.n_atoms)
>> print('Residues: ', u.residues.n_residues)
>> print('Segments: ', u.segments.n_segments)

Atoms:  1296
Residues:  1
Segments:  1

At this point however, I would like to add in topology information manually that is missing in my data file, but is important for my analysis.  I have been able to do things like add names/resnames at the atom level:

>> atoms = u.atoms

>> n_residues=432

>> u.add_TopologyAttr('name', ['O', 'H1', 'H2']*n_residues)
>> u.add_TopologyAttr('resnames', ['H2O'])

Atom Names:  ['O' 'H1' 'H2' ... 'O' 'H1' 'H2']
Atom Resnames:  ['H2O' 'H2O' 'H2O' ... 'H2O' 'H2O' 'H2O']

However, I have been unsuccessful in trying to edit anything at the residue level.  In this case, every 3 atoms (O H1 H2) is a water molecule, and I'd like to be able to identify each water as such.  Currently there is only 1 resid:

>> u.residues.resids
>> array([1])

But in this case I'd like to add more in.  I only have water molecules in the system, so adding u.add_TopologyAttr('resnames', ['H2O']) is correct in associating each atom with a water residue, however I need to be able to refer to each water molecule individually.  Eventually, I will need to do this dynamically on a per-frame basis by identifying via distance and angles which 2 hydrogens are associated with each oxygen, but at this point static analysis based on the input data file is enough.

Any help is appreciated, and please let me know if I need to provide more context/information.

**Comment by orbeckst** [comment]
I can't think of a simple way to split one residue (=all your waters) into multiple residues (although perhaps someone else knows a trick). It might be easier to create an alternative universe with [`Universe.empty()`](https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.empty) based on the initial information. The User Guide has some usage examples in https://userguide.mdanalysis.org/stable/examples/constructing_universe.html 

I'd do something along the lines of
1. read your LAMMPS file as above into a universe `u1`
2. create the alternative universe `u2` with `u2 = Universe.empty(n_atoms=u2.atoms.n_atoms, ...)` (possibly using atomnames from the original universe or adding the names later to the second universe)
3. Load the trajectory file in the new universe with `u2.load_new(DATA)` (hopefully that works...)

**Reply by orbeckst** [reply]
@rszukalo did the above help? If so, can you mark it as an answer as this would help others browsing the forum. Otherwise, can you state what problem remains?
